relay-general/src/store/normalize.rs

```diff
@@ -66,6 +66,22 @@ fn is_valid_platform(platform: &str) -> bool {
     VALID_PLATFORMS.contains(&platform)
 }
 
+pub fn normalize_dist(dist: &mut Option<String>) {
+    let mut erase = false;
+    if let Some(val) = dist {
+        if val.is_empty() {
+            erase = true;
+        }
+        let trimmed = val.trim();
+        if trimmed != val {
+            *val = trimmed.to_string()
+        }
+    }
+    if erase {
+        *dist = None;
+    }
+}
```

Comment:
Added the erase flag here because I cannot set `dist` while checking its contents. I don't like it though - any suggestions welcome.